File: data/gpt-4o/readable_format_HPEC/cpp/file042.cpp
#include <stdio.h>

double* newSxx;
double* newSyy;
int len = 1000;

void foo(double* newSxx, double* newSyy, int len);

int main()
{
 newSxx = new double[len];
 newSyy = new double[len];

 foo(newSxx, newSyy, len);

 if(newSxx != nullptr)
 {
  delete[] newSxx;
  newSxx = nullptr;
 }

 if(newSyy != nullptr)
 {
  delete[] newSyy;
  newSyy = nullptr;
 }

 return 0;
}

void foo(double* newSxx, double* newSyy, int len)
{
 // function implementation
}

File: data/deepseek_coder_base_6.7B/readable_format_HPEC/fortran/file042.f90
program DRB067_restrictpointer1_orig_no
    use omp_lib
    use DRB067
    implicit none

    integer :: len = 1000
    integer, parameter :: dp = kind(1.0d0)
    real(dp),dimension(:), pointer :: newSxx, newSyy

    allocate (newSxx(len))
    allocate (newSyy(len))

    call foo(newSxx, newSyy, len)

    if(associated(newSxx))nullify(newSxx)
    if(associated(newSyy))nullify(newSyy)
end program
